librz/core/cmd_debug.c

```diff
@@ -5036,6 +4408,602 @@ RZ_IPI int rz_cmd_debug(void *data, const char *input) {
+	for (int i = 1; i < argc; i++) {
+		if (!rz_bp_plugin_del(core->dbg->bp, argv[i])) {
+			RZ_LOG_ERROR("Failed to delete breakpoint plugin handler: %s\n", argv[i]);
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbt
+RZ_IPI RzCmdStatus rz_cmd_debug_display_bt_handler(RzCore *core, int argc, const char **argv) {
+	RzList *list = rz_debug_frames(core->dbg, UT64_MAX);
+	if (!list) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int i = 0;
+	RzListIter *iter;
+	RzDebugFrame *frame;
+	rz_list_foreach (list, iter, frame) {
+		char *flagdesc, *flagdesc2, *pcstr, *spstr;
+		get_backtrace_info(core, frame, UT64_MAX, &flagdesc, &flagdesc2, &pcstr, &spstr);
+		RzAnalysisFunction *fcn = rz_analysis_get_fcn_in(core->analysis, frame->addr, 0);
+		rz_cons_printf("%d  %s sp: %s  %-5d"
+			       "[%s]  %s %s\n",
+			i++, pcstr, spstr, (int)frame->size, fcn ? fcn->name : "??", flagdesc ? flagdesc : "", flagdesc2 ? flagdesc2 : "");
+		free(flagdesc);
+		free(flagdesc2);
+		free(pcstr);
+		free(spstr);
+	}
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbt*
+RZ_IPI RzCmdStatus rz_cmd_debug_display_bt_flags_handler(RzCore *core, int argc, const char **argv) {
+	RzList *list = rz_debug_frames(core->dbg, UT64_MAX);
+	if (!list) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int i = 0;
+	RzListIter *iter;
+	RzDebugFrame *frame;
+	rz_list_reverse(list);
+	rz_cons_printf("f-bt.*\n");
+	rz_list_foreach (list, iter, frame) {
+		rz_cons_printf("f bt.frame%d = 0x%08" PFMT64x "\n", i, frame->addr);
+		rz_cons_printf("f bt.frame%d.stack %d 0x%08" PFMT64x "\n", i, frame->size, frame->sp);
+		i++;
+	}
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbt=
+RZ_IPI RzCmdStatus rz_cmd_debug_display_bt_oneline_handler(RzCore *core, int argc, const char **argv) {
+	int mode = 0;
+	if (argc > 1) {
+		if (!strcmp(argv[1], "b")) {
+			mode = 1;
+		} else if (!strcmp(argv[1], "s")) {
+			mode = 2;
+		}
+	}
+	RzList *list = rz_debug_frames(core->dbg, UT64_MAX);
+	if (!list) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int i = 0;
+	RzListIter *iter;
+	RzDebugFrame *frame;
+	rz_list_reverse(list);
+	rz_list_foreach (list, iter, frame) {
+		if (i != 0) {
+			rz_cons_printf(" ");
+		}
+		switch (mode) {
+		case 0:
+			rz_cons_printf("0x08%" PFMT64x, frame->addr);
+			break;
+		case 1:
+			rz_cons_printf("0x08%" PFMT64x, frame->bp);
+			break;
+		case 2:
+			rz_cons_printf("0x08%" PFMT64x, frame->sp);
+			break;
+		}
+	}
+	rz_cons_newline();
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbtv
+RZ_IPI RzCmdStatus rz_cmd_debug_display_bt_local_vars_handler(RzCore *core, int argc, const char **argv) {
+	RzList *list = rz_debug_frames(core->dbg, UT64_MAX);
+	if (!list) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	backtrace_vars(core, list);
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dbtj
+RZ_IPI RzCmdStatus rz_cmd_debug_display_bt_json_handler(RzCore *core, int argc, const char **argv) {
```

Comment:
I am not 100% sure, but some of these modes, default one (`dbt`) and JSON(`dbtj`) and be combined together by using `RzCmdStateOutput *state` argument and `type: RZ_CMD_DESC_TYPE_ARGV_STATE` in the YAML.
`dbt*` is probably `RZ_OUTPUT_MODE_RIZIN`

Reply:
Yes, you are right. But, would that be useful, since the lines of code will stay the same, and the help would be less understandable (because it would squash `dbt`, `dbtj`, `dbt*` into a single command with a single description)?

Reply:
its fine - its already the case for many commands like that